### PR TITLE
Do pipeline parallelism tests in double because TF32 environment variables can be painful to manage across test suites

### PR DESCRIPTION
**Pull request by eqy**
CC @Aidyn-A @crcrpar

### CODE REVIEW

**Comment by Aidyn-A** [comment]
Awesome! Looks good to me :+1:

File: tests/L0/run_transformer/test_pipeline_parallel_fwd_bwd.py [hunk 1]
```diff
@@ -50,10 +50,10 @@ def init_weights(m):
 
 def get_target_loss_and_model(global_batch_shape: tuple, hidden_size: int, total_layers: int) -> float:
```
Comment (Aidyn-A) [inline tests/L0/run_transformer/test_pipeline_parallel_fwd_bwd.py:51]: Sorry, this was my mistake, but could you please change the returning type

File: tests/L0/run_transformer/test_pipeline_parallel_fwd_bwd.py [hunk 2]
```diff
@@ -48,12 +48,12 @@ def init_weights(m):
     return init_weights
 
 
-def get_target_loss_and_model(global_batch_shape: tuple, hidden_size: int, total_layers: int) -> float:  
+def get_target_loss_and_model(global_batch_shape: tuple, hidden_size: int, total_layers: int) -> Tuple[float, List[torch.nn.Module]]:
```
Comment (crcrpar) [inline tests/L0/run_transformer/test_pipeline_parallel_fwd_bwd.py:51]: I guess `[w, b]` should be `List[torch.Tensor]`?
```suggestion
def get_target_loss_and_model(global_batch_shape: tuple, hidden_size: int, total_layers: int) -> Tuple[float, List[torch.Tensor]]: 
```

**Review by crcrpar — APPROVED** [review]

File: tests/L0/run_transformer/test_pipeline_parallel_fwd_bwd.py (hunk 2 above)
Comment (Aidyn-A) [reply tests/L0/run_transformer/test_pipeline_parallel_fwd_bwd.py:51]: I shall add that `loss` is a `torch.Tensor` of single element.
